packages/cf-deploy-config-sub-generator/src/app/index.ts

```diff
@@ -268,6 +269,10 @@ export default class extends DeploymentGenerator {
 
         if (!this.launchDeployConfigAsSubGenerator) {
             await this._writing();
+        } else {
+            // Need to delay `init` as the yaml configurations wont be ready!
+            await this._init();
+            await this._writing();
```

Comment:
While I'm not against moving it here I'm wondering how this fixes anything?

Reply:
I've no idea! It only impacts the mta writing/reading via the mta binary. I've a feeling, based on the other lifecycle metods, that something is changing i.e. working root directory or node_path that the globally installed `mta` binary can't be found which is why it tries to lookup locally inside the `commonLibs` directory where it will never exist.

Reply:
Also, from other conversations, we shouldnt be writing in the end phase. With all the latest changes, we haven't been able to do this before but it seems after some local testing, the init and writing phases can now be moved to an earlier lifecycle method.